packages/react/src/ReactContext.js

```diff
@@ -11,8 +11,24 @@ import {REACT_PROVIDER_TYPE, REACT_CONTEXT_TYPE} from 'shared/ReactSymbols';
 
 import type {ReactContext} from 'shared/ReactTypes';
 
+import invariant from 'shared/invariant';
 import warningWithoutStack from 'shared/warningWithoutStack';
 
+import ReactCurrentOwner from './ReactCurrentOwner';
+
+export function readContext<T>(
+  context: ReactContext<T>,
+  observedBits: void | number | boolean,
+): T {
+  const dispatcher = ReactCurrentOwner.currentDispatcher;
+  invariant(
+    dispatcher !== null,
```

Comment:
This should probably do a typeof function check since this is essentially external and hostile.